Add troubleshooting section about mixed local and non local installations (#752)
napari/docs: docs/troubleshooting.md

See [napari#7288](https://github.com/napari/napari/issues/7288).

### Mixed napari installations

Since napari 0.6.2 we have added a check for the mix of [editable](https://setuptools.pypa.io/en/latest/userguide/development_mode.html) and non-editable napari installations in the same environment.
It means the situation when you install napari using `pip install -e path/to/napari` while still some napari related files are 

> present in regular installation directory (called `site-packages`).

Reviewer: ```suggestion
This occurs when you install napari using `pip install -e path/to/napari` while napari related files are still 
present in the typical conda installation directory (called `site-packages`).
```

> Because of how importing in python works, it often leads to a crash on startup or unexpected behavior when 
> these two installations are from, even slightly, different versions of napari.

Reviewer: ```suggestion
Because of how importing in python works, two installations from even slightly different versions of napari
will often lead to a crash on startup or other unexpected behavior
```

If you meet an exception starting from `RuntimeError: Mix of local and non local installation detected.` you have two options: 

> 1. Recreate your environment and install napari from scratch. It is the safest option. 

Reviewer: ```suggestion
1. Recreate your environment and install napari from scratch; this is the safest option. 
```
@Czaki, a note that I'm changing many of your 'It' to 'This'. You are grammatically correct by using 'It', but 'This' would be more typically used for this level of writing. In actuality, I try to avoid pronouns like 'this' or 'it' generally, but in this case the referred object is very clear.

   But you need to know which packages you need to install in your environment. You may use `pip freeze` to get the list of packages installed in your environment.
2. Remove the non-editable napari installation from your environment. 
   In the exception message you will see the path to the non-editable installation. 
   For example `Path to a napari directory: /home/czaki/.pyenv/versions/3.12.9/envs/napari_3.12/lib/python3.12/site-packages/napari.`

>    You can fix it by manually deleting the directory.

Reviewer: ```suggestion
   You can fix the error by manually deleting the `napari` directory.
```

>    In some rare situations it still may not fix the environment, and you may need to recreate it.

Reviewer: ```suggestion
   In some rare situations this solution may not fix the environment,
   so you may need to recreate, as described in the first option.
```
